tools/twix/src/panels/image_color_select.rs

```diff
@@ -0,0 +1,282 @@
+                Ok(cycler) => {
+                    error!("Invalid vision cycler: {cycler}");
+                    None
+                }
+                Err(error) => {
+                    error!("{error}");
+                    None
+                }
+            })
+            .unwrap_or(Cycler::VisionTop);
+        let output = CyclerOutput {
+            cycler,
+            output: Output::Main {
+                path: "image".to_string(),
+            },
+        };
+        let image_buffer = nao.subscribe_output(output);
+        let cycler_selector = VisionCyclerSelector::new(cycler);
+
+        let brush_size = 50.0;
+        Self {
+            nao,
+            image_buffer,
+            cycler_selector,
+            brush_size,
+        }
+    }
+
+    fn save(&self) -> Value {
+        let cycler = self.cycler_selector.selected_cycler();
+        json!({
+            "cycler": cycler.to_string(),
+        })
+    }
+}
+
+impl Widget for &mut ImageColorSelectPanel {
+    fn ui(self, ui: &mut Ui) -> Response {
+        ui.horizontal(|ui| {
+            if self.cycler_selector.ui(ui).changed() {
+                let output = CyclerOutput {
+                    cycler: self.cycler_selector.selected_cycler(),
+                    output: Output::Main {
+                        path: "image".to_string(),
+                    },
+                };
+                self.image_buffer = self.nao.subscribe_output(output);
+            }
+            ui.add(egui::Slider::new(&mut self.brush_size, 1.0..=200.0).text("Brush"));
+        });
+
+        ui.separator();
+
+        let image = match self.get_image() {
+            Ok(image) => image,
+            Err(error) => {
+                return ui.label(format!("{error:#?}"));
+            }
+        };
+
+        let handle = ui
+            .ctx()
+            .load_texture("image", image.clone(), TextureOptions::default())
+            .id();
+        let texture = SizedTexture {
+            id: handle,
+            size: Vec2::new(image.width() as f32, image.height() as f32),
+        };
+        let response = ui.image(texture);
+
+        ui.separator();
+
+        let painter = TwixPainter::<Pixel>::paint_at(ui, response.rect).with_camera(
+            vector![image.width() as f32, image.height() as f32],
+            Similarity2::identity(),
+            CoordinateSystem::LeftHand,
+        );
+
+        if let Some(hoverpos) = response.hover_pos() {
+            let pixel_pos = painter.transform_pixel_to_world(hoverpos);
+            if pixel_pos.x() < image.width() as f32 && pixel_pos.y() < image.height() as f32 {
+                let scroll_delta = ui.input(|input| input.scroll_delta);
+                self.brush_size = (self.brush_size + scroll_delta[1]).clamp(1.0, 200.0);
+
+                let mut max = PixelColor {
+                    red: 0.0,
+                    green: 0.0,
+                    blue: 0.0,
+                };
```

Comment:
can you use helper functions or constructors, or constants like `PixelColor::BLACK`?